Add closed_date, unassign user, and close activity log entry to reports.

billyhunt opened the pull request:
[Link to ZenHub issue.](https://github.com/usdoj-crt/crt-portal-management/issues/1075)

## What does this change?

Currently, some reports are not generating closed_dates when they are closed.  After some investigation, it seems as if the bulk action is the culprit.  This PR adds the logic currently found in the report detail view close action to bulk actions.

## Screenshots (for front-end PR):

## Checklist:

It turns out that batch processing was not adding a date to reports and updating the activity log. To test

Batch close 3 reports that are currently open and assigned to someone. Make sure that all three reports are closed, have an entry in the activity log, have a closed date, and are not assigned to anyone.
Batch close 2 open reports and 1 closed report. Make sure the 2 open reports were closed and the one already closed does not have an updated closed date.
Batch close 3 reports that are not assigned to anyone. Make sure that all three reports are closed, have an entry in the activity log, have a closed date, and are not assigned to anyone.

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [x] Check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [x] Re-check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

louh commented:
I'm testing this:

> Batch close 2 open reports and 1 closed report. Make sure the 2 open reports were closed and the one already closed does not have an updated closed date.

And sometimes it doesn't work. I think you can duplicate like this:

1. setup
  - Have 1 closed report
  - Have 1 open report, assigned to someone
2. batch close these 2 reports
3. the second report is now marked closed, but does not have closed date and did not remove the unassignment.

billyhunt commented:
> I'm testing this:
> 
> > Batch close 2 open reports and 1 closed report. Make sure the 2 open reports were closed and the one already closed does not have an updated closed date.
> 
> And sometimes it doesn't work. I think you can duplicate like this:
> 
> 1. setup
> 
> * Have 1 closed report
> * Have 1 open report, assigned to someone
> 
> 2. batch close these 2 reports
> 3. the second report is now marked closed, but does not have closed date and did not remove the unassignment.

Good catch.  I have updated, and added a test to check for this moving forward.

louh reviewed (approved):
LGTM
